### PR TITLE
Remove merge flag from tokencache

### PR DESCRIPTION
**Pull request by MarkZuber**

### CODE REVIEW

File: src/Microsoft.Identity.Client/Exceptions/CoreErrorMessages.cs [hunk 1]
```diff
@@ -165,5 +165,8 @@ public static string CustomWebUiRedirectUriWasNotMatchedToProperUri(string expec
         }
 
         public const string CustomWebUiAuthorizationCodeFailed = "CustomWebUi AcquireAuthorizationCode failed";
+
+        public const string TokenCacheJsonSerializerFailedParse = "MSAL V3 Deserialization failed to parse the cache contents.  Is this possibly an earlier format needed for DeserializeMsalV2?";
```
Comment (jmprieur) [inline src/Microsoft.Identity.Client/Exceptions/CoreErrorMessages.cs]: Can we please add in both strings: "(See https://aka.ms/msal-net-3x-cache-breaking-change)"

File: src/Microsoft.Identity.Client/TokenCache.cs [hunk 2]
```diff
@@ -1087,12 +1087,10 @@ public byte[] SerializeAdalV3()
         /// otherwise just use <see cref="SerializeMsalV3"/>/<see cref="DeserializeMsalV3"/>. 
         /// </summary>
         /// <param name="adalV3State">Array of bytes containing serialized Adal.NET V3 cache data</param>
-        /// <param name="merge">Future: Boolean enabling merging the cache into the existing cache.</param>
         /// <remarks>
         /// <see cref="SerializeMsalV3"/>/<see cref="DeserializeMsalV3"/> is compatible with other MSAL libraries such as MSAL for Python and MSAL for Java.
-        /// Merge = true is not yet available. This will be available in a later version of MSAL.NET.
         /// </remarks>
```
Comment (henrik-me) [inline src/Microsoft.Identity.Client/TokenCache.cs]: ></remarks> [](start = 12, length = 10)

For all deserialization methods, I suggest that there is a remark saying that the cache is merged into the existing cache ? Perhaps adding a few caveats to the comments?

File: src/Microsoft.Identity.Client/Exceptions/CoreErrorMessages.cs (hunk 1 above)
Comment (henrik-me) [inline src/Microsoft.Identity.Client/Exceptions/CoreErrorMessages.cs]: >  [](start = 127, length = 1)

nit: one space too many (same below)

Comment (henrik-me) [inline src/Microsoft.Identity.Client/Exceptions/CoreErrorMessages.cs]: >TokenCacheJsonSerializerFailedParse [](start = 28, length = 35)

suggestion: Deserialization failed to parse the content. Your data might be of a different format (See: DeserializeMsalV3, DeserializeMsalV2, DeserializeAdalV3).

File: src/Microsoft.Identity.Client/TokenCache.cs (hunk 2 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Client/TokenCache.cs]: adalV3State: Array of bytes containing serialized Adal.NET V3 cache data to be merged into the in memory cache?

Comment (MarkZuber) [reply src/Microsoft.Identity.Client/TokenCache.cs]: adal v3 state isn't merged due to the nature of the cache.  i'm going to leave documentation here as is and we can discuss and do a follow up PR.

---
In reply to: [261829599](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/927#discussion_r261829599) [](ancestors = 261829599)

**Pull request merged**